yolact/index.py

```diff
@@ -0,0 +1,369 @@
+                        help='Detections with a score under this threshold will not be considered. This currently only works in display mode.')
+    parser.add_argument('--dataset', default=None, type=str,
+                        help='If specified, override the dataset specified in the config with this one (example: coco2017_dataset).')
+    parser.add_argument('--detect', default=False, dest='detect', action='store_true',
+                        help='Don\'t evauluate the mask branch at all and only do object detection. This only works for --display and --benchmark.')
+    parser.add_argument('--display_fps', default=False, dest='display_fps', action='store_true',
+                        help='When displaying / saving video, draw the FPS on the frame')
+    parser.add_argument('--emulate_playback', default=False, dest='emulate_playback', action='store_true',
+                        help='When saving a video, emulate the framerate that you\'d get running in real-time mode.')
+
+    parser.set_defaults(no_bar=False, display=False, resume=False, output_coco_json=False, output_web_json=False, shuffle=False,
+                        benchmark=False, no_sort=False, no_hash=False, mask_proto_debug=False, crop=True, detect=False, display_fps=False,
+                        emulate_playback=False)
+
+    global args
+    args = parser.parse_args(argv)
+
+    if args.output_web_json:
+        args.output_coco_json = True
+
+    if args.seed is not None:
+        random.seed(args.seed)
+
+def evaluate(net:Yolact, dataset, train_mode=False):
+    net.detect.use_fast_nms = args.fast_nms
+    net.detect.use_cross_class_nms = args.cross_class_nms
+    cfg.mask_proto_debug = args.mask_proto_debug
+    nn_index = NearestNeighbourIndex('indices')
+    if args.images is not None:
+        nn_index.index_images(net, args.images)
+        return
+if __name__ == '__main__':
+    parse_args()
+
+    if args.config is not None:
+        set_cfg(args.config)
+
+    if args.trained_model == 'interrupt':
+        args.trained_model = SavePath.get_interrupt('weights/')
+    elif args.trained_model == 'latest':
+        args.trained_model = SavePath.get_latest('weights/', cfg.name)
+
+    if args.config is None:
+        model_path = SavePath.from_str(args.trained_model)
+        # TODO: Bad practice? Probably want to do a name lookup instead.
+        args.config = model_path.model_name + '_config'
+        print('Config not specified. Parsed %s from the file name.\n' % args.config)
+        set_cfg(args.config)
+
+    if args.detect:
+        cfg.eval_mask_branch = False
+
+    with torch.no_grad():
+        if not os.path.exists('results'):
+            os.makedirs('results')
+
+        if args.cuda:
+            cudnn.fastest = True
+            torch.set_default_tensor_type('torch.cuda.FloatTensor')
+        else:
+            torch.set_default_tensor_type('torch.FloatTensor')
+
+        dataset = None
```

Comment:
Do you set dataset = None here, because we are not evaluating the dataset, but instead seeing the results of the trained model?

Reply:
so, dataset is None as it's only doing inference on stored images on a folder